### PR TITLE
DOC: JBatch guide for issue 2785

### PR DESCRIPTION
**Pull request by ljamen**
Replaces PR #3596 for JBatch documentation.

### CODE REVIEW

File: docs/mp/guides/39_jbatch.adoc [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+----
+
+and the following result:
+
+[source,bash]
+
+----
+{"Steps executed":"[step1, step2]","Status":"COMPLETED"}
+----
+
```
Comment (dalexandrov) [inline docs/mp/guides/39_jbatch.adoc:321]: Please add the following:

To check job status, simply call:

[source,bash]
----
 curl -X GET http://localhost:8080/batch/status/1 (1)
----
(1) Job Id may differ.

and you should see the following output:

[source,bash]
----
{"Steps executed":"[step1, step2]","Status":"COMPLETED"}
----

**Review by dalexandrov — CHANGES_REQUESTED** [review]
Hello Lisa, 
there were some changes in the JBatch example. I have requested changes to the guide to follow those changes. Could you be so kind to apply them :)

Thank you :)

File: docs/mp/guides/39_jbatch.adoc [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+NOTE: You must specify the fully qualified names in the `ref` properties, like “io.helidon.jbatch.example.jobs.MyItemReader”, otherwise it will not work.
+
+== Create an Endpoint
+Create a small endpoint to activate the job:
+
+[source,java]
+.new endpoint
+
+----
```
[the remark this answers is not in the document]
Comment (ljamen) [reply docs/mp/guides/39_jbatch.adoc:230]: @dalexandrov Do you have the complete code example in your examples file? I think there may be a copy paste issue with this example. I want to take it directly from your source.

**Comment by dalexandrov** [comment]
Hello Lisa,
this is the full example: https://github.com/oracle/helidon/tree/helidon-2.x/examples/jbatch

File: docs/mp/guides/39_jbatch.adoc [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+==== 1. Create a unit of input information
+
+[source,java]
+.MyInputRecord
+----
+public class MyInputRecord {
+    private int id;
+
+        public MyInputRecord(int id) {
```
Comment (dalexandrov) [inline docs/mp/guides/39_jbatch.adoc:64]: Please fix Indentation

File: docs/mp/guides/39_jbatch.adoc [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+}
+----
+
+Helidon specifies to JBatch that it should run in Standalone (SE) mode. It will also register the `HelidonExecutorServiceProvider` which is actually relatively small. For our example we need something really small, like a `FixedTheadPool` with 2 threads. This provider is used to tell our JBatch engine exactly which ExecutorSevice to use.
+
+[source, java]
+.HelidonExecutorServiceProvider
+
+----
+public class HelidonExecutorServiceProvider implements ExecutorServiceProvider {
```
Comment (dalexandrov) [inline docs/mp/guides/39_jbatch.adoc:283]: Please fix the indentation

**Review by dalexandrov — CHANGES_REQUESTED** [review]

File: docs/mp/guides/39_jbatch.adoc [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
```
Comment (dalexandrov) [inline docs/mp/guides/39_jbatch.adoc:281]: Should be:

```
public class HelidonExecutorServiceProvider implements ExecutorServiceProvider {
    @Override
    public ExecutorService getExecutorService() {
        return ThreadPoolSupplier.builder().corePoolSize(2).build().get();
    }
}
```

Comment (dalexandrov) [inline docs/mp/guides/39_jbatch.adoc:281]: Should be:

```
public class HelidonExecutorServiceProvider implements ExecutorServiceProvider {
    @Override
    public ExecutorService getExecutorService() {
        return ThreadPoolSupplier.builder().corePoolSize(2).build().get();
    }
}
```

**Review by dalexandrov — CHANGES_REQUESTED** [review]

**Review by dalexandrov — APPROVED** [review]

**Pull request merged**